### PR TITLE
Update logging to use key value pairs

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context

Using the Godaddy logging package we can now use key value pairs
with .with(). This PR updates all logging to use key value pairs.

# What has changed

updates all logging to use key value pairs.

# How to test?

Ran ITs and ATs.
Make sure logs lines have not been missed.

# Links

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/notify/client/LoggingNotificationClient.java [hunk 1]
```diff
@@ -24,56 +24,68 @@ public LoggingNotificationClient(NotificationClientApi client, Logger log) {
   public SendEmailResponse sendEmail(
       String templateId, String emailAddress, Map<String, String> personalisation, String reference)
       throws NotificationClientException {
-    log.debug("sendEmail: {} - {} - {} - {}", templateId, emailAddress, personalisation, reference);
+    log.with("template_id", templateId)
+        .with("personalisation", personalisation)
+        .with("reference", reference)
+        .debug("sendEmail");
```
Comment (tej99) [inline src/main/java/uk/gov/ons/ctp/response/notify/client/LoggingNotificationClient.java]: send_email

File: src/main/java/uk/gov/ons/ctp/response/notify/endpoint/StatusEndpoint.java [hunk 2]
```diff
@@ -106,8 +106,7 @@ private Notification fetchNotificationFromNotifyService(final UUID notificationI
       return notifyService.findNotificationById(notificationId);
     } catch (NotificationClientException e) {
       String errorMsg = String.format(ERRORMSG_NOTIFICATION_ISSUE, e.getMessage(), e.getCause());
```
Comment (tej99) [inline src/main/java/uk/gov/ons/ctp/response/notify/endpoint/StatusEndpoint.java]: Do we still need `e.getMessage(), e.getCause()`?

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/ctp/response/notify/endpoint/StatusEndpoint.java]: well it goes into the exception that gets thrown

Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/response/notify/endpoint/StatusEndpoint.java]: But we're passing the whole exception through anyway `log.error(errorMsg, e);`

Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/response/notify/endpoint/StatusEndpoint.java]: Oh just realised it's getting used somewhere else, nm

**Pull request merged**